tests/generic_config_updater/test_dynamic_acl.py

```diff
@@ -0,0 +1,1246 @@
+        priority -= 1
+        rule_number += 1
+
+    tmpfile = generate_tmpfile(duthost)
+    logger.info("tmpfile {}".format(tmpfile))
+
+    try:
+        output = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        expect_op_success(duthost, output)
+
+        for rule_name, expected_content in expected_rule_contents.items():
+            expect_acl_rule_match(duthost, rule_name, expected_content)
+
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+
+def dynamic_acl_remove_ip_forward_rule(duthost, ip_type):
+    """Remove selected forward rule from the acl table"""
+
+    if ip_type == "IPV4":
+        rule_name = "RULE_1"
+    else:
+        rule_name = "RULE_2"
+
+    extra_vars = {
+        "rule_name": rule_name
+    }
+
+    output = format_and_apply_template(duthost, REMOVE_RULE_TEMPLATE, extra_vars)
+
+    expect_op_success(duthost, output)
+
+    expect_acl_rule_removed(duthost, rule_name)
+
+
+def dynamic_acl_remove_table(duthost):
+    """Remove an ACL Table Type from the duthost"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_TABLE_FILE)
+
+    expect_op_success(duthost, output)
+
+
+def dynamic_acl_remove_nonexistent_table(duthost):
+    """Remove a nonexistent ACL Table from the duthost, verify it fails"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_NONEXISTENT_TABLE_FILE)
+
+    expect_op_failure(output)
+
+
+def dynamic_acl_remove_table_type(duthost):
+    """Remove an ACL Table definition from the duthost"""
+
+    output = load_and_apply_json_patch(duthost, REMOVE_TABLE_TYPE_FILE)
+
+    expect_op_success(duthost, output)
+
+
+def test_gcu_acl_arp_rule_creation(rand_selected_dut,
+                                   ptfadapter,
+                                   setup,
+                                   dynamic_acl_create_table,
+                                   arp_packets_for_test,
+                                   ip_and_intf_info,
+                                   toggle_all_simulator_ports_to_rand_selected_tor):  # noqa F811
+    """Test that we can create a blanket ARP packet forwarding rule with GCU, and that ARP packets
+    are correctly forwarded while all others are dropped"""
+
+    ptf_intf_ipv4_addr, _, ptf_intf_ipv6_addr, _, ptf_intf_index, port_name = ip_and_intf_info
+
+    ip_version, outgoing_packet, expected_packet = arp_packets_for_test
+
+    dynamic_acl_create_arp_forward_rule(rand_selected_dut)
+    dynamic_acl_create_secondary_drop_rule(rand_selected_dut, setup, port_name)
+
+    if ip_version == 'v4':
+        pytest_require(ptf_intf_ipv4_addr is not None, 'No IPv4 VLAN address configured on device')
+    elif ip_version == 'v6':
+        pytest_require(ptf_intf_ipv6_addr is not None, 'No IPv6 VLAN address configured on device')
+
+    ptfadapter.dataplane.flush()
+    testutils.send(ptfadapter, ptf_intf_index, outgoing_packet)
+    testutils.verify_packet(ptfadapter, expected_packet, ptf_intf_index, timeout=10)
```

Comment:
I didn't see the code for adding IP in the ptf interface. Can you please point it out?

Reply:
I may be mistaken, but I thought it was done here in the arp test setup fixture:

` intf1_indice = mg_facts['minigraph_ptf_indices'][intf1]`

`asic.config_ip_intf(intf1,  vlan_ipv4_subnet, "add")`

Reply:
Regarding the ARP test, the ARP request can't be send by PtfAdapter because ARP request is control plane packet. We can follow below steps
1. Configure IPv4/IPv6 address (must be in the vlan subnet) on 1 of the pft port
2. Apply the drop rules and forwarding rules
3. Clear ARP cache from the device with command `sonic-clear arp`
4. Trigger ARP/NDP with ping command on DUT.
5. Check ARP (for IPv4) and neighbor (for IPv6) is learnt with command `show arp` or ` nbrshow  -6`